src/Domain/Block.cpp

```diff
@@ -0,0 +1,52 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "Domain/Block.hpp"
+
+#include <ostream>
+
+#include "Utilities/StdHelpers.hpp"
+
+template <size_t VolumeDim>
+Block<VolumeDim>::Block(
+    const EmbeddingMap<VolumeDim, VolumeDim>& embedding_map, const size_t id,
+    std::unordered_map<Direction<VolumeDim>, BlockNeighbor<VolumeDim>>&&
+        neighbors)
+    : embedding_map_(embedding_map.get_clone()),
+      id_(id),
+      neighbors_(std::move(neighbors)) {
+  // Loop over Directions to search which Directions were not set to neighbors_,
+  // set these Directions to external_boundaries_.
+  for (size_t j = 0; j < 2 * VolumeDim; j++) {
+    // This is a map from [0,1,2,3,4,5] to [0,0,1,1,2,2], with 0-xi, 1-eta,
+    // 2-zeta:
+    size_t dimension = (j - j % 2) / 2;
+    Side side;
+    // With even/odd-ness designating lower/upper:
+    side = j % 2 == 1 ? Side::Upper : Side::Lower;
+    Direction<VolumeDim> input_direction(dimension, side);
+    if (neighbors_.find(input_direction) == neighbors_.end()) {
+      external_boundaries_.emplace(std::move(input_direction));
+    }
+  }
+}
+
+template <size_t VolumeDim>
+std::ostream& operator<<(std::ostream& os, const Block<VolumeDim>& block) {
+  os << "Block " << block.id() << ":\n";
+  os << "Neighbors:\n";
+  for (const auto& kv : block.neighbors()) {
```

Comment:
Would `key_value` help? In this case we might want to actually have `firstName_secondName`